pontoon/api/tests/test_views.py

```diff
@@ -0,0 +1,675 @@
+        )
+
+    assert response.status_code == 200
+    assert response.data == {
+        "slug": "project_a",
+        "name": "Project A",
+        "priority": 1,
+        "deadline": None,
+        "visibility": "public",
+        "contact": None,
+        "info": "",
+        "system_project": False,
+        "disabled": False,
+        "sync_disabled": False,
+        "pretranslation_enabled": False,
+        "total_strings": 25,
+        "approved_strings": 15,
+        "pretranslated_strings": 0,
+        "strings_with_warnings": 2,
+        "strings_with_errors": 3,
+        "missing_strings": 5,
+        "unreviewed_strings": 5,
+        "complete": False,
+        "tags": [],
+        "localizations": {},
+    }
+
+
+@pytest.mark.django_db
+def test_regular_projects(
+    django_assert_num_queries,
+):
+    ProjectFactory.create_batch(3, disabled=True)
+    ProjectFactory.create_batch(3, system_project=True)
+    with django_assert_num_queries(4):
+        response = APIClient().get("/api/v2/projects/")
+
+    assert response.status_code == 200
+
+    expected_results = [
+        {
+            "slug": p.slug,
+            "name": p.name,
+            "priority": p.priority,
+            "deadline": p.deadline,
+            "visibility": p.visibility,
+            "contact": p.contact,
+            "info": p.info,
+            "system_project": p.system_project,
+            "disabled": p.disabled,
+            "sync_disabled": p.sync_disabled,
+            "pretranslation_enabled": p.pretranslation_enabled,
+            "total_strings": p.total_strings,
+            "approved_strings": p.approved_strings,
+            "pretranslated_strings": p.pretranslated_strings,
+            "strings_with_warnings": p.strings_with_warnings,
+            "strings_with_errors": p.strings_with_errors,
+            "missing_strings": p.missing_strings,
+            "unreviewed_strings": p.unreviewed_strings,
+            "complete": p.complete,
+            "tags": [
+                tag.name for tag in sorted(p.tags.all(), key=lambda tag: tag.name)
+            ],
+            "locales": [
+                locale.code
+                for locale in sorted(p.locales.all(), key=lambda locale: locale.code)
+            ],
+        }
+        for p in sorted(
+            Project.objects.filter(disabled=False, system_project=False),
+            key=lambda p: p.pk,
+        )
+    ]
+
+    for project in response.data["results"]:
+        project["locales"] = sorted(project["locales"])
+        project["tags"] = sorted(project["tags"])
+
+    results = sorted(response.data["results"], key=lambda p: p["slug"])
+    expected_results = sorted(expected_results, key=lambda p: p["slug"])
+
+    # includes Terminology project
+    assert response.data["count"] == 1
+    assert results == expected_results
+
+
+@pytest.mark.django_db
+def test_all_projects_stats(django_assert_num_queries):
```

Comment:
This and the `test_projects()` above should be just one test case, same as we joined test cases for locales.

Reply:
Oh, good point. I think you can remove `test_all_projects_stats()` and move stats test to `test_projects`, because we already separatelly test system and disabled projects.